adapters/repos/db/lsmkv/segment_blockmax.go

```diff
@@ -156,12 +156,35 @@ type SegmentBlockMax struct {
 	sectionReader *io.SectionReader
 }
 
+func generateSingleFilter(tombstones *sroar.Bitmap, filterDocIds helpers.AllowList) (*sroar.Bitmap, *sroar.Bitmap) {
+	if tombstones != nil && tombstones.GetCardinality() == 0 {
```

Comment:
`GetCardinality` does some calculations internally every time it is called. In this case `IsEmpty` could be used to avoid calculating exact value